libqrexec/exec.c

```diff
@@ -236,15 +242,27 @@ static int find_file(
 
         if (path_length + name_length + 1 >= buffer_size) {
             LOG(ERROR, "find_qrexec_service_file: buffer too small for file path");
-            return -1;
+            return -2;
```

Comment:
Too long argument for a file path should still continue the search for service file without argument.

Reply:
This should never happen with the default search path.  Names longer than `NAME_MAX` are rejected earlier: https://github.com/QubesOS/qubes-core-qrexec/blob/6787a57af73558f949b33a6479d0ba451d078593/libqrexec/exec.c#L232-L233

So this really means that either you have a really long entry in the search path, or there is a bug somewhere.

Reply:
What about service+arg being for example NAME_MAX-1 long?